### PR TITLE
A bunch of readability and documentation attempts

### PR DESCRIPTION
**Pull request by asn-d6**
Did a first pass of the code and suggested some improvements here!

Missing items for a future PR:
- More robust unittests
- README
- Blend in the trusted setup from https://github.com/asn-d6/kzgverify so that we can test bigger matrices in the tests

### CODE REVIEW

File: my_types.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+"""
+A bunch of type definitions to improve readability
```
Comment (adietrichs) [inline my_types.py:2]: might be simpler to just have that in shared.py?

File: verifier.py [hunk 2]
```diff
@@ -1,76 +1,109 @@
...
-        if aggregated_column_data[j] == [0] * N_locs:
+        if aggregated_column_samples[j] == [0] * N_locs: # skip this column if we have no samples from it
             continue
-        coset_factor = get_coset_factor(j, N_locs)
-        interpolation_polynomial = fft(list_to_reverse_bit_order(aggregated_column_data[j]), MODULUS, root_of_unity, True)
-        interpolation_polynomial = [div(c, pow(coset_factor, i, MODULUS)) for i, c in enumerate(interpolation_polynomial)]
-        aggregated_interpolation_polynomial = field_multiadd(aggregated_interpolation_polynomial, interpolation_polynomial)
+        coset_factor = get_coset_factor(j, N_locs) # get coset factor for this column
+
+        # Get interpolation polynomial for this column. To do so we first do an FFT over the roots of unity and then we
```
Comment (adietrichs) [inline verifier.py]: IFFT or IDFT would be more precise than FFT

**Review by adietrichs — COMMENTED** [review]
left 2 small comments, otherwise looks good to merge.

File: verifier.py (hunk 2 above)
Comment (asn-d6) [reply verifier.py]: Fixed in b5b2234

File: my_types.py (hunk 1 above)
Comment (asn-d6) [reply my_types.py:2]: I attempted that, but there is a cyclic dependency between `shared.py` and `setup.py` which makes things messy.

See 8286a2f for an attempt (but I had to duplicate the types in `setup.py`). I can revert if you don't like.

**Comment by asn-d6** [comment]
Pushed some fixes to the review!

File: my_types.py (hunk 1 above)
Comment (adietrichs) [reply my_types.py:2]: Ah, I see the problem now. On second thought it does indeed seem cleaner to keep the typing separate in its own file then. Would you mind reverting that last commit?

**Comment by adietrichs** [comment]
lgtm, merging.

**Pull request merged**